File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length;
        int [] result = new int[n];
        
        result[0] = 1;
        for(int i=1; i<=n-1; i++){
            result[i] = result[i-1] * nums[i-1];
        }
        
        int rightproduct = 1;
        
        for(int i=n-1; i>=0; i--){
            result[i] *= rightproduct;
            rightproduct *= nums[i];
        }
        return result;
        
    }
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        answer =[1 for _ in range(len(nums))]
        n = len(nums)
        
        for i in range(1, n):
            answer[i] = answer[i-1]* nums[i-1]
            
        rightproduct =1
        
        for i in range(n-1, -1, -1 ):
            answer[i] *= rightproduct;
            rightproduct *= nums[i];
        
        return answer 